### PR TITLE
Prevent search input caret position to reset on edits

### LINKED ISSUE #1124: Search input does not handle inputs properly
1. Go to https://explorer.near.org
2. Type anything (e.g. "near")
3. Move cursor to the left (e.g. between "n" and "e")
4. Hit [DELETE] / [BACKSPACE] button or some other character
5. Observe that the cursor jumps to the end of the input, so it is not possible to add several characters in the middle of the string or remove 2+ characters from the middle of the string

### PR DESCRIPTION
**Pull request by kristoferlund**
Resolves #1124

The caret position is no longer reset when edits inside the search string is made. Came across this bug while Saturday GitHub browsing and thought I might as well fix it since it was so small. Change of query param now only sets search input value on first page load.

### CODE REVIEW

**Comment by kristoferlund** [comment]
Nudge @luixo @shelegdmitriy

File: frontend/src/components/utils/Search.tsx [hunk 1]
```diff
@@ -182,10 +183,18 @@ const Search: React.FC<Props> = React.memo(({ dashboard }) => {
...
   const track = useAnalyticsTrack();
 
-  const [value, setValue] = useQueryParam("query");
+  const [queryParamQuery, setQueryParamQuery] = useQueryParam("query");
+  const [value, setValue] = React.useState<string>("");
   const [searchValue, setSearchValue] = React.useState<string | undefined>(
     undefined
   );
+
+  React.useEffect(() => {
```
Comment (luixo) [inline frontend/src/components/utils/Search.tsx:192]: This should be replaces with initialValue as `queryParamQuery` (let's name it `queryValue`) as initial value of `value`'s `useState`.

**Comment by luixo** [comment]
Closed in favor of https://github.com/near/near-explorer/pull/1154

**Comment by kristoferlund** [comment]
> Closed in favor of #1154

Ah, cool, did not have time to revisit this one to fix the changes you requested.

**Pull request closed without merging**